mlflow/pyspark/ml/__init__.py

```diff
@@ -0,0 +1,202 @@
+import logging
+from pkg_resources import resource_filename
+
+import mlflow
+from mlflow.utils import chunk_dict, truncate_dict
+from mlflow.utils.annotations import experimental
+from mlflow.utils.autologging_utils import (
+    autologging_integration,
+    safe_patch,
+    try_mlflow_log,
+)
+
+from mlflow import spark as mlflow_spark
+
+from pyspark.ml.param import Params
+
+
+_logger = logging.getLogger(__name__)
+
+
+def _get_fully_qualified_class_name(instance):
+    return instance.__class__.__module__ + "." + instance.__class__.__name__
+
+
+def _read_log_model_allowlist():
+    """
+    Reads the module allowlist and returns it as a set.
+    """
+    file_path = resource_filename(__name__, "log_model_allowlist.txt")
+    allowlist = set()
+    with open(file_path) as f:
+        for line in f:
+            stripped = line.strip()
+            is_blankline_or_comment = stripped == "" or stripped.startswith("#")
+            if not is_blankline_or_comment:
+                allowlist.add(stripped)
+    return allowlist
+
+
+_log_model_allowlist = _read_log_model_allowlist()
+
+
+def _should_log_model(estimator, spark_model):
+    class_name = _get_fully_qualified_class_name(estimator)
+    if class_name in _log_model_allowlist:
+        if class_name == 'pyspark.ml.classification.LogisticRegression':
+            return spark_model.numClasses < 10
+        else:
+            return True
+    else:
+        return False
+
+
+def _get_estimator_info_tags(estimator):
+    """
+    :return: A dictionary of MLflow run tag keys and values
+             describing the specified estimator.
+    """
+    return {
+        "estimator_name": estimator.__class__.__name__,
+        "estimator_class": _get_fully_qualified_class_name(estimator),
+    }
+
+
+def _get_instance_param_map(instance):
+    param_map = {param.name: instance.getOrDefault(param)
+                 for param in instance.params if instance.isDefined(param)}
+    # TODO:
+    #  handle special case: CrossValidator.estimatorParamMaps
+
+    expanded_param_map = {}
+    for k, v in param_map.items():
+        if isinstance(v, Params):
+            expanded_param_map[k] = v.uid
+            internal_param_map = _get_instance_param_map(v)
+            for ik, iv in internal_param_map.items():
+                expanded_param_map[f'{v.uid}.{ik}'] = iv
+        else:
+            expanded_param_map[k] = v
```

Comment:
Do we need special hanling for `Pipeline.stages`?

Reply:
Yes. We need check all stages in PipelineModel.stages are in allowlist. I will add then in followup PR when adding support for pipeline